ns/log.n3

```diff
@@ -0,0 +1,396 @@
+
+#log:Falsehood a rdf:Class;
+#   rdfs:comment "Something which is not true. Not the complement of True!";
+#   rdfs:domain log:context.
+#
+
+log:implies a rdf:Property;
+    rdfs:comment """Logical implication. 
+This is the relation between the antecedent (subject) and
+conclusion (object) of a rule.
+The application of a rule to a knowledge-base is as follows.
+For every substitution which, applied to the antecedent,
+gives a formula which is a subset of the knowledge-base,
+then the result of applying that same substitution to the
+conclusion may be added to the knowledge-base.
+
+related: See log:conclusion. 
+(See the CWM manual for command line options to determine how
+rules from different sources are applied to and the results
+added to various formula.)
+""";
+    rdfs:domain log:Formula;
+    rdfs:range log:Formula;
+    rdfs:isDefinedBy <>.
+
+log:includes a rdf:Property;
+    rdfs:comment """The subject formula includes the object formula.
+Formula A includes formula B if there exists some substitution
+which when applied to B creates a formula B' such that for
+every statement in B' is also in A, every variable
+universally (or existentially) quantified in B' is quantified in
+the same way in A. 
+
+Variable substitution is applied recursively to nested compound terms such as
+formulae, lists and sets.
+
+(Understood natively by cwm when in in the antecedent of a rule.
+You can use this to peer inside nested formulae.)""";
+    rdfs:domain log:Formula;
+    rdfs:range log:Formula;
+    rdfs:isDefinedBy <>.
+
+log:notIncludes a rdf:Property;
+    rdfs:comment """The object formula is NOT a subset of subject.
+True iff log:includes is false.
+The converse of log:includes. (Understood natively by cwm.
+The subject formula may contain variables.
+(In cwm, variables must of course end up getting bound
+before the log:include test can be done, or an infinite result set
+would result)
+Related: See includes""";
+    rdfs:domain log:Formula;
+    rdfs:range log:Formula;
+    rdfs:isDefinedBy <>.
+
+
+# The following connect the inference engine to the web
+
+
+log:parsedAsN3 a rdf:Property;
+    rdfs:comment "The subject string, parsed as N3, gives this formula.";
+    rdfs:label "parses to";
+    rdfs:domain log:String;
+    rdfs:range log:Formula.
+ 
+ log:n3String a rdf:Property;
+    rdfs:comment "The subject formula, expressed as N3, gives this string.";
+    rdfs:label "n3 string";
+    rdfs:domain log:Formula;
+    rdfs:range log:String.
+
+log:semantics a rdf:Property;
+    rdfs:domain doc:Work;
+    rdfs:range  log:Formula;
+    rdfs:label  "semantics";
+    rdfs:comment """The log:semantics of a document is the formula.
+achieved by parsing representation of the document.
+For a document in Notation3, log:semantics is the
+log:parsedAsN3 of the log:content of the document.
+For a document in another RDF syntax, it is parsed according to the
+appropriate specification to yield an RDF formula
+(a subclass of N3 log:Formula).
```

Comment:
Updated to be generic about other RDF syntaxes, not just RDF/XML.

Reply:
Personally I'd leave a caveat here, as to not force N3 systems to support all RDF syntaxes. Something along the lines of "if the RDF syntax is supported, .. , else an error should be raised (?)"

Reply:
It's much like the linked-data platform, where there are one or two syntaxes required by the spec, but content-negotiation allows for other formats as well. A processor _should_ process alternate formats, and _must_ evaluate to **false** on failure. In addition to Notation3, formats required to be supported include N-Triples and Turtle?

Reply:
To me yes but in EYE we use --turtle <uri> to get N-Triples or Turtle data:
```
Usage: eye <options>* <data>* <query>*
...
<data>
    [--n3] <uri>                    N3 triples and rules
    --pl <uri>                      Prolog facts and rules
    --proof <uri>                   N3 proof lemmas
    --turtle <uri>                  Turtle triples
...
```

Reply:
That doesn't scale to other formats (RDF/XML, RDFa, JSON-LD, TriG, ...); we could consider a form with a list subject where additional values are interpreted like an HTTP Accept header to indicate preference. We could also describe a default Accept header (similar to how JSON-LD does) to use to allow the response to be content-negotiated.

We should probably also describe how/(if) an RDF Dataset can be turned into a Formula with recursively quoted graphs.

Reply:
@gkellogg 
> We should probably also describe how/(if) an RDF Dataset can be turned into a Formula with recursively quoted graphs.

I'd leave that for later - see the [note in the spec](https://w3c.github.io/N3/spec/#rel-n3-trig)